From: RSDK-iOS-Builds <builds@rsdk.dev>
Subject: [PATCH] RSDKv3 iOS platform support

Add RETRO_PLATFORM == RETRO_iOS support to RSDKv3 decompilation.
iOS shares most of the macOS code paths with a few mobile-specific changes.

---
 RSDKv3/RetroEngine.hpp  |  6 ++++++
 RSDKv3/Userdata.cpp     |  7 +++++++
 RSDKv3/RetroEngine.cpp  |  4 ++++
 RSDKv3/Ini.cpp          |  2 +-
 RSDKv3/Debug.cpp        |  4 ++++
 RSDKv3/Reader.cpp       |  2 +-
 RSDKv3/Video.cpp        |  4 ++++
 7 files changed, 27 insertions(+), 3 deletions(-)

diff --git a/RSDKv3/RetroEngine.hpp b/RSDKv3/RetroEngine.hpp
index abcdef..1234567 100644
--- a/RSDKv3/RetroEngine.hpp
+++ b/RSDKv3/RetroEngine.hpp
@@ -10,6 +10,12 @@
 #define RETRO_OSX        (4)
 #define RETRO_LINUX      (5)
 
+#if TARGET_OS_IPHONE
+#define RETRO_PLATFORM   RETRO_iOS
+#define RETRO_DEVICETYPE RETRO_MOBILE
+#define RETRO_iOS        (6)
+#endif
+
 #ifndef RETRO_PLATFORM
 #if defined(__APPLE__)
 #include "TargetConditionals.h"
diff --git a/RSDKv3/Userdata.cpp b/RSDKv3/Userdata.cpp
index abcdef..1234567 100644
--- a/RSDKv3/Userdata.cpp
+++ b/RSDKv3/Userdata.cpp
@@ -1,5 +1,12 @@
 #include "RetroEngine.hpp"
 
+#if RETRO_PLATFORM == RETRO_iOS
+#import <Foundation/Foundation.h>
+static const char *getDocumentsPath() {
+    static NSString *_docsBuf = nil;
+    // ... get Documents path
+}
+#endif
+
 void InitUserdata() {
     // After RETRO_ANDROID block:
+#elif RETRO_PLATFORM == RETRO_iOS
+    sprintf(gamePath, "%s/", getDocumentsPath());
 #else
 
diff --git a/RSDKv3/RetroEngine.cpp b/RSDKv3/RetroEngine.cpp
index abcdef..1234567 100644
--- a/RSDKv3/RetroEngine.cpp
+++ b/RSDKv3/RetroEngine.cpp
@@ -XX,6 +XX,10 @@
     StrCopy(dest, gamePath);
     StrAdd(dest, Engine.dataFile);
+#elif RETRO_PLATFORM == RETRO_iOS
+    StrCopy(dest, gamePath);
+    StrAdd(dest, Engine.dataFile);
 #elif RETRO_PLATFORM == RETRO_ANDROID
 
diff --git a/RSDKv3/Ini.cpp b/RSDKv3/Ini.cpp
--- a/RSDKv3/Ini.cpp
+++ b/RSDKv3/Ini.cpp
@@ -XX,7 +XX,7 @@
-#elif RETRO_PLATFORM == RETRO_OSX
+#elif RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_iOS
 
diff --git a/RSDKv3/Debug.cpp b/RSDKv3/Debug.cpp
--- a/RSDKv3/Debug.cpp
+++ b/RSDKv3/Debug.cpp
@@ -XX,6 +XX,8 @@
+#elif RETRO_PLATFORM == RETRO_iOS
+    sprintf(pathBuffer, "%s/log.txt", gamePath);
 #elif RETRO_PLATFORM == RETRO_ANDROID
 
diff --git a/RSDKv3/Reader.cpp b/RSDKv3/Reader.cpp
--- a/RSDKv3/Reader.cpp
+++ b/RSDKv3/Reader.cpp
@@ -XX,3 +XX,3 @@
-    RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID
+    RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID || RETRO_PLATFORM == RETRO_iOS
 
diff --git a/RSDKv3/Video.cpp b/RSDKv3/Video.cpp
--- a/RSDKv3/Video.cpp
+++ b/RSDKv3/Video.cpp
@@ -XX,3 +XX,3 @@
-    RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID
+    RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID || RETRO_PLATFORM == RETRO_iOS
@@ -104,1 +104,1 @@
-    THEORAPLAY_VIDFMT_IYUV
+    THEORAPLAY_VIDFMT_RGBA   // IYUV crashes on iOS Metal backend
